packages/synapse-react-client/src/components/SetBasicAccessRequirementFields/SetBasicAccessRequirementFields.tsx

```diff
@@ -53,32 +62,39 @@ export const SetBasicAccessRequirementFields = React.forwardRef(
       }
     }, [accessRequirement])
 
-    const { mutate: updateAccessRequirement, error: updateArError } =
-      useUpdateAccessRequirement<BasicAccessRequirement>({
-        onSuccess: updatedAr => {
-          if (isDeletingTextInstructions) {
-            setUpdatedAr(updatedAr)
-            setIsDeletingTextInstructions(false)
-          } else {
+    const { mutateAsync: updateAccessRequirement, error: updateArError } =
+      useUpdateAccessRequirement<BasicAccessRequirement>()
+
+    const onUpdateAccessRequirement = useCallback(
+      async (
+        updatedAr: BasicAccessRequirement,
+        isDeletingTextInstructions: boolean,
+      ) => {
+        try {
```

Comment:
Use `mutateAsync`, so `isDeletingTextInstructions` doesn't need to be stored in state, and catch any errors, since react query won't internally catch and discard the error for us as discussed [here](https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync)